src/lib/mojaloop-requests/wso2auth.js

```diff
@@ -77,12 +75,15 @@ class WSO2Auth {
         try {
             const response = await request(reqOpts);
             this.token = response.access_token;
-            this.refreshSeconds = Math.min(this.refreshSeconds, response.expires_in - 5);
+            const tokenExpiry = response.expires_in > 0 ? response.expires_in : Infinity;
+            this.refreshSeconds = Math.min(this.refreshSeconds * 1000, tokenExpiry) / 1000;
```

Comment:
I think you need more guard conditions around the value in the returned token. If, for example you get a string, you will end up doing something like setTimeout(..., NaN); which I believe will result in setTimeout(...,1); I know the spec says it will be an integer, but we could be dealing with any implementation of OAuth2.0, We cannot assume the returned token will be valid.

Reply:
done